### DISCUSSION TITLE
[Question][Hw3] About the least element of CPO

### DISCUSSION THREAD

**Discussion by kohs100** [prosyslab-classroom/cs524-program-analysis#296] [Q&A] [answered]
Name: Hyungseok Ko
Possible related question: #31 

While trying to prove CPOness in HW3 Problem 2, I found that the definition and lemma for CPO in lecture 3 slide 25 seems counterintuitive for me.

![image](https://github.com/prosyslab-classroom/cs524-program-analysis/assets/20630805/826d925d-2a87-4049-9041-dd5556076fc0)

I understand that, since every element in the poset can be upper bound of empty set, $\bigsqcup\emptyset$ is eventually translated as the least element in given poset.

It seems natural that following poset is also CPO, since its all possible chains have $\top$ as its LUB.
$$(\mathbb{Z}_\top, \sqsubseteq)$$
where
$$\sqsubseteq=\\{ (n, \top) \mid n \in \mathbb{Z} \\}$$
(Identity relation omitted)
However, it does not have a single least element like the lemma states. Is this kind of problem is just matter of dualism, or do I have any misunderstanding for the concept?

**Comment by KihongHeo** [comment]
Come to class! Let me explain.

**Comment by kohs100** [comment, marked as answer]
In class explanation: Since $\emptyset$ is both subset of $D$ and total ordered, $\emptyset$ also satisfies the definition of chain. So $\emptyset$ should also have LUB in $D$ to satisfy the definition of CPO.

Thus the given set $(\mathbb{Z}_\top,\sqsubseteq)$ is not CPO because it does not have the least element for every element in $D$.

Thank you for the explanation, professor.
